stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/querychannels/internal/QueryChannelsDatabaseLogic.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.offline.plugin.logic.querychannels.internal
+
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.ChannelConfig
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.query.QueryChannelsSpec
+import io.getstream.chat.android.client.query.pagination.AnyChannelPaginationRequest
+
+internal interface QueryChannelsDatabaseLogic {
```

Comment:
This internal interface is not needed.

Reply:
The interfaces are not necessary right now, but once those classes are moved to other modules, the interface will be necessary to access the implementation of the interfaces from another module. At some point, the `QueryChannelLogic` will have to be moved away from state module and the interfaces will have to be moved together (visibility must also be changed), but not the implementations

Reply:
When we need to move it to another module, we can create an Interface if it is needed